### PR TITLE
sparse state prep: allow user to pick target bitsize if needed

### PR DESCRIPTION
**Pull request by anurudhp**
- user can pass a target bitsize instead of always using log(N)
- normalize the coeff vector passed to `StatePreparationViaRotations`

### CODE REVIEW

**Comment by anurudhp** [comment]
@tanujkhattar ptal

File: qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations.py [hunk 1]
```diff
@@ -58,6 +58,7 @@ class SparseStatePreparationViaRotations(Bloq):
     nonzero_coeffs: Union[tuple[complex, ...], HasLength] = field(converter=_to_tuple_or_has_length)
     N: SymbolicInt
     phase_bitsize: SymbolicInt
+    target_bitsize: SymbolicInt = field()
```
Comment (tanujkhattar) [inline qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations.py:61]: Can you add a test that uses a bigger target_bitsize than default and passes?

Comment (anurudhp) [reply qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations.py:63]: added a bloq example and test

**Review by tanujkhattar — APPROVED** [review]

File: qualtran/conftest.py [hunk 2]
```diff
@@ -116,6 +116,7 @@ def assert_bloq_example_serializes_for_pytest(bloq_ex: BloqExample):
         'state_prep_via_rotation_symb',  # cannot serialize HasLength
         'state_prep_via_rotation_symb_phasegrad',  # cannot serialize Shaped
         'sparse_state_prep_via_rotations',  # cannot serialize Permutation
+        'sparse_state_prep_via_rotations_with_large_target_bitsize',  # setting an array element with a sequence.
```
Comment (tanujkhattar) [inline qualtran/conftest.py:119]: What's going on here? Is it because we can't serialize permutation?

**Review by tanujkhattar — APPROVED** [review]
LGTM % nits

File: qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations_test.py [hunk 3]
```diff
@@ -20,12 +20,17 @@
...
 
 
-def test_examples(bloq_autotester):
-    bloq_autotester(_sparse_state_prep_via_rotations)
+@pytest.mark.parametrize(
+    "bloq_ex",
+    [_sparse_state_prep_via_rotations, _sparse_state_prep_via_rotations_with_large_target_bitsize],
+)
+def test_examples(bloq_autotester, bloq_ex):
+    bloq_autotester(bloq_ex)
```
[the remark this answers is not in the document]
Comment (anurudhp) [reply qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations_test.py:36]: added simulation tests (at the end)

[the remark this answers is not in the document]
Comment (tanujkhattar) [reply qualtran/bloqs/state_preparation/sparse_state_preparation_via_rotations_test.py:36]: Thanks

**Pull request merged**